lua/lib/fs.lua

```diff
@@ -109,6 +113,92 @@ local function remove_dir(cwd)
   return luv.fs_rmdir(cwd)
 end
 
+local function do_copy(source, destination)
+  local source_stats = luv.fs_stat(source)
+
+  if source_stats and source_stats.type == 'file' then
+    return luv.fs_copyfile(source, destination)
+  end
+
+  local handle = luv.fs_scandir(source)
+
+  if type(handle) == 'string' then
+    return false, handle
+  end
+
+  luv.fs_mkdir(destination, source_stats.mode)
+
+  while true do
+    local name, t = luv.fs_scandir_next(handle)
+    if not name then break end
+
+    local new_name = source..'/'..name
+    local new_destination = destination..'/'..name
+    local success, msg = do_copy(new_name, new_destination)
+    if not success then return success, msg end
+  end
+
+  return true
+end
+
+local function do_paste(node, action_type, action_fn)
+  local clip = clipboard[action_type]
+  if #clip == 0 then return end
+
+  local destination = node.absolute_path
+  local stats = luv.fs_stat(destination)
+  local is_dir = stats and stats.type == 'directory'
+
+  if not is_dir then
+    destination = vim.fn.fnamemodify(destination, ':p:h')
+  elseif not node.open then
+    destination = vim.fn.fnamemodify(destination, ':p:h:h')
+  end
+
+  local msg = #clip..' entries'
+
+  if #clip == 1 then
+    msg = clip[1].absolute_path
+  end
+
+  local ans = vim.fn.input(action_type..' '..msg..' to '..destination..'? y/n: ')
+  clear_prompt()
+  if not ans:match('^y') then
+    return api.nvim_out_write('Canceled.\n')
+  end
+
+  for _, entry in ipairs(clip) do
+    local dest = destination..'/'..entry.name
+    local dest_stats = luv.fs_stat(dest)
+    if dest_stats then
+      local ans = vim.fn.input(dest..' already exists, overwrite ? y/n: ')
+      clear_prompt()
+      if not ans:match('^y') then goto continue end
+    end
+
+    local success, msg = action_fn(entry.absolute_path, dest)
+    if not success then
+      api.nvim_err_writeln('Could not '..action_type..' '..entry.absolute_path..' - '..msg)
+    end
+    ::continue::
```

Comment:
goto is not available for some distributions of neovim (as neovim uses lua 5.1 where goto is not available). You could leave it for now as i still have some piece of code with `goto`s but this issue will need to be addressed in another PR.

Reply:
I changed the logic to not use continue.